subcommand/webhook-cert-manager/command_test.go

```diff
@@ -0,0 +1,532 @@
+
+	ctx := context.TODO()
+	timer := &retry.Timer{Timeout: 10 * time.Second, Wait: 500 * time.Millisecond}
+	// First, check that the original secret contents are updated when the cert-manager starts.
+	retry.RunWith(timer, t, func(r *retry.R) {
+		secret1, err := k8s.CoreV1().Secrets("default").Get(ctx, secretOne, metav1.GetOptions{})
+		require.NoError(r, err)
+		require.NotEqual(r, secret1.Data[v1.TLSCertKey], []byte("cert-1"))
+		require.NotEqual(r, secret1.Data[v1.TLSPrivateKeyKey], []byte("private-key-1"))
+
+		certificate = secret1.Data[v1.TLSCertKey]
+		key = secret1.Data[v1.TLSPrivateKeyKey]
+
+		webhookConfig1, err := k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(ctx, webhookConfigOne, metav1.GetOptions{})
+		require.NoError(r, err)
+		require.NotEqual(r, webhookConfig1.Webhooks[0].ClientConfig.CABundle, caBundleOne)
+	})
+
+	// Wait for certs to be rotated.
+	time.Sleep(2 * time.Second)
+
+	timer = &retry.Timer{Timeout: 10 * time.Second, Wait: 500 * time.Millisecond}
+	// Check that the certificate is rotated and the secret is updated.
+	retry.RunWith(timer, t, func(r *retry.R) {
+		secret1, err := k8s.CoreV1().Secrets("default").Get(ctx, secretOne, metav1.GetOptions{})
+		require.NoError(r, err)
+		require.NotEqual(r, secret1.Data[v1.TLSCertKey], certificate)
+		require.NotEqual(r, secret1.Data[v1.TLSPrivateKeyKey], key)
+	})
+}
+
+// This test verifies that when there is an error while attempting to update
+// the certs or the webhook config, it retries the update every second until
+// it succeeds.
+func TestCertWatcher(t *testing.T) {
+	t.Parallel()
+
+	webhookName := "webhookOne"
+	webhook := &admissionv1beta1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookName,
+		},
+		Webhooks: []admissionv1beta1.MutatingWebhook{
+			{
+				Name:         "webhook-under-test",
+				ClientConfig: admissionv1beta1.WebhookClientConfig{},
+			},
+		},
+	}
+
+	k8s := fake.NewSimpleClientset(webhook)
+	ui := cli.NewMockUi()
+
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+
+	file, err := ioutil.TempFile("", "config.json")
+	require.NoError(t, err)
+	defer os.Remove(file.Name())
+
+	_, err = file.Write([]byte(configFileUpdates))
+	require.NoError(t, err)
+
+	exitCh := runCommandAsynchronously(&cmd, []string{
+		"-config-file", file.Name(),
+	})
+	defer stopCommand(t, &cmd, exitCh)
+
+	ctx := context.TODO()
+	// This first sleep is required to ensure the command had completed validating that the MWC does exist in the cluster
+	// It ensures the MWC is absent when the server actually attempts to update it with a bundle.
+	time.Sleep(1 * time.Millisecond)
+	err = k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Delete(ctx, webhookName, metav1.DeleteOptions{})
+	require.NoError(t, err)
+
+	// This sleep ensures the cluster attempts to try updating the MWC every second and fails before we re-create it in the cluster.
+	time.Sleep(2 * time.Second)
```

Comment:
Can you play with the timeouts so we don't need to wait this long? Can we set the error retry to 100ms?